### PR TITLE
Fuzz a couple of `Deserialize` implementations

### LINKED ISSUE #2765: Fuzz `Deserialize` implementations
Would have caught #2764 without me trying this by hand.

### PR DESCRIPTION
**Pull request by hrxi**
`Bitset`, `TrieNode` and `KeyNibbles`. I've also made sure that the respective implementations look sane.

I also checked that the fuzzing would have caught #2764, the fuzzer found these crashes immediately.

Some more implementations don't need fuzzing, I added some comments for that.

Fixes #2765.

### CODE REVIEW

File: fuzz/src/bin/00_testcases.rs
Comment (nibhar) [inline fuzz/src/bin/00_testcases.rs:1]: Weird name for a file? What do the `00` stand for?

**Review by nibhar — APPROVED** [review]
Cool stuff!
Would these be supposed to be run continuously in our CI? Making sure nothing breaks?

**Review by jsdanielh — APPROVED** [review]

**Comment by hrxi** [comment]
> Would these be supposed to be run continuously in our CI? Making sure nothing breaks?

That sounds like a good idea, but not implemented in this PR unfortunately.

File: fuzz/src/bin/00_testcases.rs
Comment (hrxi) [reply fuzz/src/bin/00_testcases.rs:1]: It's a file different from the others. The 00 is so that it's sorted first. ^^

**Pull request merged**